### PR TITLE
ufunc tests

### PR DESCRIPTION
**Pull request by samjrholt**

### CODE REVIEW

File: discretisedfield/field.py [hunk 1]
```diff
@@ -3790,27 +3790,36 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
...
             return None
         else:
-            return self.__class__(
-                mesh[0],
-                nvdim=result.shape[-1],
-                value=result,
-                vdims=self.vdims,
-                vdim_mapping=self.vdim_mapping,
-            )
+            print("Hi")
```
Comment (swapneelap) [inline discretisedfield/field.py:3810]: Maybe remove this 😉

File: discretisedfield/field.py [hunk 2]
```diff
@@ -3790,27 +3790,36 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
...
+            if not np.array_equal(result.shape[:-1], mesh[0].n):
```
Comment (swapneelap) [inline discretisedfield/field.py:3811]: Can we also do it like this?
```suggestion
            if not np.array_equal(result.shape[:-1], self.mesh.n):
```

File: discretisedfield/field.py [hunk 3]
```diff
@@ -3790,27 +3790,36 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
...
+                raise NotImplementedError()
+            try:
+                return self.__class__(
+                    mesh[0],
```
Comment (swapneelap) [inline discretisedfield/field.py:3815]: Here as well, can we do this?
```suggestion
                    self.mesh,
```

**Review by swapneelap — COMMENTED** [review]
Looks good to me. I have left a few small comments.

File: discretisedfield/field.py (hunk 1 above)
Comment (samjrholt) [inline discretisedfield/field.py:3810]:
```suggestion
```

**Pull request merged**